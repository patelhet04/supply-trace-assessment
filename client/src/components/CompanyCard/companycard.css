.company-card {
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 100px;
  border: 1px solid #ccc;
  border-radius: 6px;
  background-color: white;
  box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
}

.company-card-details {
  padding: 1rem;
}
.company-card-details header {
  color: #000;
  margin: 0;
  font-size: 1.1rem;
  font-weight: bold;
}

.company-card-details p {
  margin: 0.5rem 0 0;
  color: #666;
  font-size: 1rem;
}

@media (min-width: 768px) and (max-width: 1199px) {
  .company-card-details header {
    font-size: 1rem;
  }
  .company-card-details p {
    font-size: 0.9rem;
  }
}
