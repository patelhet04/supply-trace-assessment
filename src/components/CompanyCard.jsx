import React from "react";

const CompanyCard = ({ companyData }) => {
  return <div>CompanyCard</div>;
};

export default CompanyCard;
