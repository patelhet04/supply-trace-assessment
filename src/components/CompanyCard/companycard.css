.company-card {
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 100px;
  border: 1px solid #ccc;
  border-radius: 8px;
  background-color: white;
  box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
}

.company-card-details {
  padding: 1rem;
}
.company-card-details h3 {
  color: #000;
  margin: 0;
  font-size: 1.2rem;
}

.company-card-details p {
  margin: 0.5rem 0 0;
  color: #666;
}
