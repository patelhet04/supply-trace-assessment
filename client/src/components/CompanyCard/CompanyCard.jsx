import React from "react";
import PropTypes from "prop-types";
import "./companycard.css";

// Functional component to display company details
const CompanyCard = ({ companyData }) => {
  return (
    <div className="company-card">
      <div className="company-card-details">
        <header>{companyData.name}</header>
        <p>{companyData.address}</p>
      </div>
    </div>
  );
};

// PropTypes for type checking
CompanyCard.propTypes = {
  companyData: PropTypes.shape({
    company_id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    address: PropTypes.string.isRequired,
    latitude: PropTypes.number,
    longitude: PropTypes.number,
  }).isRequired,
};

export default CompanyCard;
