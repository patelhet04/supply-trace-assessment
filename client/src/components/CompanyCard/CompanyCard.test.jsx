import React from "react";
import { render } from "@testing-library/react";
import CompanyCard from "./CompanyCard";

const mockCompanyData = {
  company_id: 1,
  name: "TechNova Solutions",
  address: "123 Innovation Drive, San Francisco, CA 94105",
  latitude: 37.7749,
  longitude: -122.4194,
};

describe("CompanyCard", () => {
  it("renders the company name", () => {
    const { getByText } = render(<CompanyCard companyData={mockCompanyData} />);
    expect(getByText("TechNova Solutions")).toBeInTheDocument();
  });

  it("renders the company address", () => {
    const { getByText } = render(<CompanyCard companyData={mockCompanyData} />);
    expect(
      getByText("123 Innovation Drive, San Francisco, CA 94105")
    ).toBeInTheDocument();
  });

  it("matches the snapshot", () => {
    const { asFragment } = render(
      <CompanyCard companyData={mockCompanyData} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
