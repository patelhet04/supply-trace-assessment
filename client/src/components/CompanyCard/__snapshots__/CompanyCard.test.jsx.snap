// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyCard matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="company-card"
  >
    <div
      class="company-card-details"
    >
      <header>
        TechNova Solutions
      </header>
      <p>
        123 Innovation Drive, San Francisco, CA 94105
      </p>
    </div>
  </div>
</DocumentFragment>
`;
